optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/move/MoveSelectorFactoryTest.java

```diff
@@ -0,0 +1,302 @@
+                .isInstanceOf(CachingMoveSelector.class)
+                .isNotInstanceOf(ShufflingMoveSelector.class);
+        assertThat(moveSelector.getCacheType()).isEqualTo(SelectionCacheType.STEP);
+        assertThat(((CachingMoveSelector) moveSelector).getChildMoveSelector()).isSameAs(baseMoveSelector);
+    }
+
+    @Test
+    void justInTimeOriginal() {
+        final MoveSelector baseMoveSelector = SelectorTestUtils.mockMoveSelector(DummyMove.class);
+        DummyMoveSelectorConfig moveSelectorConfig = new DummyMoveSelectorConfig();
+        MoveSelectorFactory moveSelectorFactory =
+                new AssertingMoveSelectorFactory(moveSelectorConfig, baseMoveSelector, SelectionCacheType.JUST_IN_TIME, false);
+        moveSelectorConfig.setCacheType(SelectionCacheType.JUST_IN_TIME);
+        moveSelectorConfig.setSelectionOrder(SelectionOrder.ORIGINAL);
+        MoveSelector moveSelector = moveSelectorFactory.buildMoveSelector(
+                buildHeuristicConfigPolicy(),
+                SelectionCacheType.JUST_IN_TIME, SelectionOrder.RANDOM);
+        assertThat(moveSelector).isSameAs(baseMoveSelector);
+        assertThat(moveSelector.getCacheType()).isEqualTo(SelectionCacheType.JUST_IN_TIME);
+    }
+
+    @Test
+    void phaseRandom() {
+        final MoveSelector baseMoveSelector = SelectorTestUtils.mockMoveSelector(DummyMove.class);
+        DummyMoveSelectorConfig moveSelectorConfig = new DummyMoveSelectorConfig();
+        MoveSelectorFactory moveSelectorFactory =
+                new AssertingMoveSelectorFactory(moveSelectorConfig, baseMoveSelector, SelectionCacheType.PHASE, false);
+        moveSelectorConfig.setCacheType(SelectionCacheType.PHASE);
+        moveSelectorConfig.setSelectionOrder(SelectionOrder.RANDOM);
+        MoveSelector moveSelector = moveSelectorFactory.buildMoveSelector(
+                buildHeuristicConfigPolicy(),
+                SelectionCacheType.JUST_IN_TIME, SelectionOrder.RANDOM);
+        assertThat(moveSelector)
+                .isInstanceOf(CachingMoveSelector.class)
+                .isNotInstanceOf(ShufflingMoveSelector.class);
+        assertThat(moveSelector.getCacheType()).isEqualTo(SelectionCacheType.PHASE);
+        assertThat(((CachingMoveSelector) moveSelector).getChildMoveSelector()).isSameAs(baseMoveSelector);
+    }
+
+    @Test
+    void stepRandom() {
+        final MoveSelector baseMoveSelector = SelectorTestUtils.mockMoveSelector(DummyMove.class);
+        DummyMoveSelectorConfig moveSelectorConfig = new DummyMoveSelectorConfig();
+        MoveSelectorFactory moveSelectorFactory =
+                new AssertingMoveSelectorFactory(moveSelectorConfig, baseMoveSelector, SelectionCacheType.STEP, false);
+        moveSelectorConfig.setCacheType(SelectionCacheType.STEP);
+        moveSelectorConfig.setSelectionOrder(SelectionOrder.RANDOM);
+        MoveSelector moveSelector = moveSelectorFactory.buildMoveSelector(
+                buildHeuristicConfigPolicy(),
```

Comment:
Is this what the code formatter does? Pretty sure it fits on one 120-char line.

Reply:
No, the formatter is not guilty; there must have been a new line already and the formatter didn't touch it, precisely as we wanted. Some of the test methods were moved from a different test class.

Good catch, thanks!